packages/runtime/src/components/page/PageHead.tsx

```diff
@@ -0,0 +1,294 @@
+  const title = baseLocalizedPage?.meta.title ?? page.meta.title
+  const description = baseLocalizedPage?.meta.description ?? page.meta.description
+  const keywords = baseLocalizedPage?.meta.keywords ?? page.meta.keywords
+  const socialImage = baseLocalizedPage?.meta.socialImage ?? page.meta.socialImage
+  const canonicalUrl = baseLocalizedPage?.seo.canonicalUrl ?? page.seo.canonicalUrl
+  const isIndexingBlocked = baseLocalizedPage?.seo.isIndexingBlocked ?? page.seo.isIndexingBlocked
+
+  const fontFamilyParamValue = useMemo(() => {
+    if (site == null) {
+      return page.fonts
+        .map(({ family, variants }) => {
+          return `${family.replace(/ /g, '+')}:${variants.join()}`
+        })
+        .join('|')
+    }
+
+    return site.googleFonts.edges
+      .filter((edge): edge is NonNullable<typeof edge> => edge != null)
+      .map(({ activeVariants, node: { family, variants } }) => {
+        const activeVariantSpecifiers = variants
+          .filter(variant =>
+            activeVariants.some(activeVariant => activeVariant.specifier === variant.specifier),
+          )
+          .map(variant => variant.specifier)
+          .join()
+
+        return `${family.replace(/ /g, '+')}:${activeVariantSpecifiers}`
+      })
+      .join('|')
+  }, [site, page])
+
+  // When the page snippets change, we run the cleanup function for any snippets
+  // that are no longer present. This process only happens in the builder
+  const previousHeadSnippets = useRef<MakeswiftPageDocument['snippets'] | null>(null)
+  useEffect(() => {
+    const headSnippetsToCleanUp = (previousHeadSnippets.current ?? [])
+      .filter(previousSnippet => previousSnippet.cleanup != null)
+      .filter(previousSnippet => !headSnippets.some(snippet => previousSnippet.id === snippet.id))
+
+    headSnippetsToCleanUp.forEach(snippetToCleanUp => {
+      if (snippetToCleanUp.cleanup == null) return
+
+      try {
+        const cleanUp = new Function(snippetToCleanUp.cleanup)
+        cleanUp()
+      } catch (err) {
+        console.error('Error attempting to parse snippet cleanup', err)
+        // Ignore errors from user input.
+      }
+    })
+
+    previousHeadSnippets.current = headSnippets
+  }, [headSnippets])
+
+  const scriptRefs = useRef<(HTMLScriptElement | null)[]>([])
+  function saveScriptRef(ref: HTMLScriptElement | null) {
+    if (ref == null) return
+    scriptRefs.current.push(ref)
+  }
+
+  const headElementsFragment = (
+    <>
+      <style data-makeswift-meta="true">
+        {`
+        html {
+            font-family: sans-serif;
+        }
+        div#__next {
+            overflow: hidden;
+        }
+        `}
+      </style>
+      {favicon && (
+        <link
+          rel="icon"
+          type={favicon.mimetype}
+          href={favicon.publicUrl}
+          data-makeswift-meta="true"
+        />
+      )}
+
+      {canonicalUrl && <link rel="canonical" href={canonicalUrl} data-makeswift-meta="true" />}
```

Comment:
And that's because they're inserted using `useServerInsertedHTML`, right? If we end up going with my idea then this won't be needed since we wouldn't be using `useServerInsertedHTML`.